pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -227,3 +238,32 @@ func KeyFunc(name, namespace string) string {
 	}
 	return namespace + "/" + name
 }
+
+// NamedPort defines the Port and Name tuple needed for services and endpoints.
+type NamedPort struct {
+	Port int32
+	Name string
+}
+
+// Str2NamedPort parses a proto:port string into a namePort struct.
+func Str2NamedPort(str string) (NamedPort, error) {
```

Comment:
Maybe we should move this to the only caller package, it does not belong to registry.